### PR TITLE
Enable the use of different filters and subfilters for PAdES signatures

### PR DESCRIPTION
**Pull request by davydsantos**
Main changes:
- adding extension point to enable different filter/subfilter combinations
- a fix on certificate generation (adding digital signature KeyUsage)
- adding a utility method to use MSCapi to test signatures

### CODE REVIEW

File: dss-pades/src/main/java/eu/europa/esig/dss/pdf/pdfbox/PdfBoxSignatureService.java [hunk 1]
```diff
@@ -277,7 +276,7 @@ public void saveDocumentIncrementally(PAdESSignatureParameters parameters, Outpu
 	}
 
 	protected COSName getSubFilter() {
-		return PDSignature.SUBFILTER_ETSI_CADES_DETACHED;
+		return null;
```
Comment (davydsantos) [inline dss-pades/src/main/java/eu/europa/esig/dss/pdf/pdfbox/PdfBoxSignatureService.java]: Leave the default sub filter value empty to enable the use of whatever the signature parameters have configured.

File: dss-test/src/main/java/eu/europa/esig/dss/test/gen/CertificateService.java [hunk 2]
```diff
@@ -99,8 +101,8 @@ public MockPrivateKeyEntry generateCertificateChain(final SignatureAlgorithm alg
 		KeyPair childKeyPair = generateKeyPair(algorithm.getEncryptionAlgorithm());
 
 		X500Name childSubject = new X500Name("CN=SignerFake,O=DSS-test");
-		CertificateToken child = generateRootCertificateWithCrl(algorithm, childSubject, rootName, rootEntry.getPrivateKey(), childKeyPair.getPublic(),
-				notBefore, notAfter);
+		CertificateToken child = generateCertificate(algorithm, childSubject, rootName, rootEntry.getPrivateKey(), childKeyPair.getPublic(),
+				notBefore, notAfter, new KeyUsage(KeyUsage.keyCertSign | KeyUsage.cRLSign | KeyUsage.digitalSignature));
```
Comment (davydsantos) [inline dss-test/src/main/java/eu/europa/esig/dss/test/gen/CertificateService.java]: KeyUsage.digitalSignature  was missing

**Comment by pvandenbroucke** [comment]
Hello Davyd,

I pushed this [change](https://github.com/esig/dss/commit/9914116e6fe8531d33a3889a1b67b5563199d07b) to allow you overriding of filter/subfilter. I prefer a harmonized way to set the values.

About dss-test : 

- did you meet any issue with generated test certificates ?

- why don't you use the MSCAPISignatureToken directly ? I didn't see any reference to the new method "readMscapiCertificate".

Regards,

Pierrick

**Comment by davydsantos** [comment]
So, I think the only think missing is the key usage (sorry I missed the MSCAPISignatureToken).

**Pull request closed without merging**